src/movies.js

```diff
@@ -1,22 +1,84 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const directorsArray = moviesArray.map(movie => movie.director)
+    return directorsArray
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    if (moviesArray.length === 0) {
+        return 0
+    }
+    const spielbergMovies = moviesArray.filter(movies => 
+        movies.director.includes ('Steven Spielberg') && movies.genre.includes ('Drama') === true)
+    return spielbergMovies.length
```

Comment:
El primer if esta bien peeeero no queda bien ponerlo porque estas preguntando por un length y devolviendo el valor que devolveria pues devuelve siempre el length y fuera! :D 

Te hago unos cambios!

```suggestion
    return moviesArray.filter(movies => 
        movies.director.includes ('Steven Spielberg') && movies.genre.includes ('Drama')).length
```